### ISSUE TITLE
[clang-format] Three-way comparison (spaceship operator) detected as clang format violation.

### ISSUE THREAD

**Issue by hruskape** [llvm/llvm-project#85773]
Minimal peace of code taken from [1]

```
struct Point
{
    int x;
    int y;
    auto operator<=>(const Point&) const = default;
    // ... non-comparison functions ...
};
```

With clang-format version 16.0.6, no problem detected.

With clang-format version 17.0.5 following error is printed. 
Tool suggested modification render code not to be possible to compile.

```
clang-format --dry-run test.cpp

./test.cpp:5:20: error: code should be clang-formatted [-Wclang-format-violations]
    auto operator<=>(const Point&) const = default;
                   ^
./test.cpp:5:21: error: code should be clang-formatted [-Wclang-format-violations]
    auto operator<=>(const Point&) const = default;
                    ^
FAILED clang-format found errors!

```

[1] https://en.cppreference.com/w/cpp/language/default_comparisons

**Comment by owenca** [comment]
It seems this has been fixed since 17.0.6 as I can't reproduce it with the default LLVM style.

**Comment by hruskape** [comment]
Thank you @owenca for hint with different style. I have checked that our project is using default style which is file. Our .clang-format containing "Standard: c++14" which seems to be root cause of problem. If I remove that option or define c++20 standard where this syntax was introduced, then format seems to be handled correctly. I think we should close this issue with this information.
